### ISSUE TITLE
Select - clear selection + search keyboard interactions

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7341]
### Expected Behavior
The clear and search should be accessible via keyboard in the Select component

### Actual Behavior
When you have both clear and search in a select the keyboard behavior doesn't work very well. If you arrow down to clear you can't get back up to search.

### URL, screen shot, or Codepen exhibiting the issue
https://codesandbox.io/p/sandbox/grommet-v2-template-forked-jwn8q2?file=%2Findex.js%3A7%2C24

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by miqh** [comment]
@jcfilben,

I've got an idea of how to address this issue, but first I'd ask for confirmation on how keyboard navigation is expected to behave for the Select component.

At present, with the Select component in focus:

- When both clear (top position) and search are enabled, <kbd>Down</kbd> focuses search, another <kbd>Down</kbd> focuses the first option. Is skipping clear desirable here? (i.e. Potential counter-intuitiveness that <kbd>Down</kbd> navigation skips clear, but <kbd>Up</kbd> navigation won't)
- When both clear (bottom position) and search are enabled, <kbd>Down</kbd> while on the last option won't move the focus to clear, so the only way to get there is via <kbd>Tab</kbd>. Is this current behaviour desirable to retain?

**Comment by jcfilben** [comment]
> When both clear (top position) and search are enabled, Down focuses search, another Down focuses the first option. Is skipping clear desirable here? (i.e. Potential counter-intuitiveness that Down navigation skips clear, but Up navigation won't)

I don't think we should skip the clear button here. I think the ideal behavior is that down arrow will focus search and then pressing down arrow again will focus the clear button.

> When both clear (bottom position) and search are enabled, Down while on the last option won't move the focus to clear, so the only way to get there is via Tab. Is this current behaviour desirable to retain?

No, I don't think the current behavior is correct. If we press down arrow while on the last option I would expect to move to the clear option. However we should maintain the behavior of pressing tab will also move to the clear button.

**Comment by jcfilben** [comment]
We should use tab only (not arrow keys) to move to and from the search
